### ISSUE TITLE
View post/page link-button is hidden

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/gutenberg#20309]
Process.
One creates a post and clicks publish. One skips checking the post.
Next time is says Update.
Usually there is a View Page in the top black toolbar. But ExactMetrics (Google Analytics Dashboard for WP (GADWP) plugin is located in the top black toolbar instead of the View Page link. 

Here is an example from a Norwegian web site.
<img width="1204" alt="Screen Shot 2020-02-19 at 05 40 48" src="https://user-images.githubusercontent.com/5323259/74802548-7fe0ab80-52da-11ea-93f1-b2276551ff50.png">

I have to click Update (Oppdater) to get the snackbar notification with the View Page text link.
<img width="1194" alt="Screen Shot 2020-02-19 at 05 43 56" src="https://user-images.githubusercontent.com/5323259/74802681-db129e00-52da-11ea-99b9-376f928c04dc.png">

Deactivating the plugin brings the View item link back.
<img width="1209" alt="Screen Shot 2020-02-19 at 05 46 33" src="https://user-images.githubusercontent.com/5323259/74802829-3b094480-52db-11ea-8293-d452f91bb27a.png">

With the plugin active there is no visible way for me to click a link to view the page without having to click the update button to get the snackbar notification visible.

NB! The above happens when creating a new post. When going back to enter I notice this:
<img width="1208" alt="Screen Shot 2020-02-19 at 05 54 05" src="https://user-images.githubusercontent.com/5323259/74803177-4610a480-52dc-11ea-8977-b2501384fb22.png">

ExactMetrics is now to the right of View Item.

**Comment by talldan** [comment]
@paaljoachim Not sure I follow completely. Is this an issue with the ExactMetrics plugin? I'd recommend writing some Steps for Reproduction.

The black top bar isn't part of the block editor, so I think either way this issue doesn't belong here in the Gutenberg repo. It should either be moved to core trac (https://core.trac.wordpress.org/) or you should let the plugin dev know.

**Comment by paaljoachim** [comment]
Hey Dan @talldan 

Your correct on both accounts. Closing this issue.

**Comment by sabernhardt** [comment]
This is on Trac now: https://core.trac.wordpress.org/ticket/53948

If we implement the first option, hiding the link until publish/save action, how could the editor remove the `hidden` class to show the link?

(Or if you think showing the link even before the URL resolves is acceptable, please comment about that on the ticket.)

**Comment by audrasjb** [comment]
Reopening the issue as a related ticket still exists on Trac.

**Comment by ndiego** [comment]
@audrasjb I was just catching up on the discussion in core Slack. I know the Trac ticket is on the 6.0 milestone, and it seems like the actual fix needs to take place in Gutenberg first. I will add this to the [6.0 Project Board](https://github.com/orgs/WordPress/projects/27) for discussion.

**Comment by talldan** [comment]
Could we update the issue title and description to use the proper format for bug reports and be clearer about the actual problem (@paaljoachim, it looks like you made this, so it'd be appreciated if you could revisit it)? Maybe borrow some of the details from trac.

I find this really report hard to follow because it sounds like a plugin issue. It's not great to force every person that reads it to also have to read all the comments in the issue as well as the trac ticket and comments there too.

**Comment by paaljoachim** [comment]
I retested and the title and content is edited.

**Comment by talldan** [comment]
Thanks!

**Comment by Mamaduka** [comment]
I don't think we can fix this in Gutenberg. Admin bar menu item would require a full page refresh to render.

**Comment by ndiego** [comment]
From a purely personal perspective, I think this is an acceptable "bug" to simply ignore. When you go through the publish flow, there are many links that a user can click on the "View Page/Post". I also work in Fullscreen Mode and never see the admin bar so that might be influencing my thoughts here. 😬

**Comment by sabernhardt** [comment]
This is probably more enhancement than bug, but I kept the type as reported due to the lack of parity with the Classic Editor. Using Classic, the View Post link has been available in the admin toolbar immediately after publishing/saving _because_ it refreshes the page and switches from New Post to the Edit Post screen.

If the link is added in the admin toolbar for the New Post screen (on the Trac ticket), I would need help with the editor side. The editor would need to check whether the link exists and then remove the `hidden` class when saving the post.

If it needs to wait until next release, I could update the `since` note to 6.1.0 instead of 6.0.0.

**Comment by talldan** [comment]
I think it could be possible, but it would probably require a bit of work in core. Gutenberg could use a react portal to the admin bar to render a component there.

It would require a couple of things:
- An HTML element to use for the portal
- The existing View Post link to be inactive when the block editor is used.

Like others mention, I don't know whether it's worth it. This bug is likely to have been present since 5.0, and it's only being reported now.

**Comment by paaljoachim** [comment]
We adapt to what is present and find work arounds.

Anyhow it might need a bigger resolution through design.
Linking in associated issues.

**Make Permalink/View Post easier to access**
https://github.com/WordPress/gutenberg/issues/30579

**Site Editor - 'Preview Site' feature.** (Linked to my comment there)
https://github.com/WordPress/gutenberg/issues/29398#issuecomment-988859368

**Comment by annezazu** [comment]
Closing this out as it hasn't gained much traction with additional reports in the two years since and since we have a "view post" option after publishing:

https://github.com/user-attachments/assets/5accd89a-836c-4c4b-9457-56e704cb155a
